### PR TITLE
deploy contoso cams

### PR DESCRIPTION
**Pull request by nickdala**
* Terraform
* Email protobuf
* azd

### CODE REVIEW

File: apps/proto/src/main/proto/email/v1/email.proto [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+option java_multiple_files = true;
+option java_outer_classname = "EmailProto";
+option java_package = "com.contoso.cams.protobuf.email.v1";
+
+message EmailRequest {
+  string email_address = 1;
+  string url_to_manual = 2;
+  int64 request_id = 3;
+}
```
Comment (coderandhiker) [inline apps/proto/src/main/proto/email/v1/email.proto:13]: Does this protocol need to have a subject/body? Even if we're auto filling it with constants, if those are going to be flexible enough to be handled by any recipient of this message, that might be preferable.

File: infra/dev/shared/terraform/modules/app-service/variables.tf [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+variable "location" {
+  type        = string
+  description = "The Azure region where all resources in this example should be created"
+}
+
+variable "contoso_webapp_options" {
+  type = object({
+    contoso_active_directory_tenant_id      = string
+    contoso_active_directory_client_id      = string
+    contoso_active_directory_client_secret  = string
```
Comment (coderandhiker) [inline infra/dev/shared/terraform/modules/app-service/variables.tf:26]: We'll have to unwind client secret usage. I'm going to find out if it's required before we ship MWA, be keeping that in mind if this configuration is pervasive, if it's easier to adopt certificates now let me know.

File: infra/dev/shared/terraform/modules/app-service/variables.tf [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    postgresql_database_url       = string
+    postgresql_database_user      = string
+    postgresql_database_password  = string
```
Comment (coderandhiker) [inline infra/dev/shared/terraform/modules/app-service/variables.tf:30]: safe secrets consideration here for future reference

File: infra/dev/shared/terraform/modules/app-service/variables.tf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    redis_host_name               = string
+    redis_port                    = number
+    redis_password                = string
```
Comment (coderandhiker) [inline infra/dev/shared/terraform/modules/app-service/variables.tf:34]: safe secrets consideration here for future reference

File: infra/dev/terraform/aca.tf [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+  secret {
+    name         = "azure-servicebus-connection-string"
+    value        = azurerm_servicebus_namespace.servicebus_namespace.default_primary_connection_string
+  }
+
+  template {
+    container {
+      name   = "email-processor-app"
+      image  = "nickdala/cams-email-processor:0.1"
```
Comment (coderandhiker) [inline infra/dev/terraform/aca.tf:25]: Do we have a dev shared ACR for our containers deployed into our testing subscription?

**Review by coderandhiker — APPROVED** [review]

File: infra/dev/shared/terraform/modules/app-service/variables.tf (hunk 4 above)
Comment (nickdala) [reply infra/dev/shared/terraform/modules/app-service/variables.tf:34]: Passwords are in the form "@Microsoft.KeyVault(SecretUri=...".  Secrets are not exposed.

File: infra/dev/terraform/aca.tf (hunk 5 above)
Comment (nickdala) [reply infra/dev/terraform/aca.tf:25]: This will be addressed in a future pull request.

**Pull request merged**